### PR TITLE
[chore] add Sumo Logic distribution

### PR DESCRIPTION
**Pull request by andrzej-stencel**
**Description:**

Adds Sumo Logic distribution to component docs.

**Link to tracking Issue:**

- https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21383

### CODE REVIEW

**Review by mx-psi — APPROVED** [review]
LGTM, would be good to have someone from Sumo give their approval here :)

**Review by swiatekm — APPROVED** [review]
Found one receiver that shouldn't be there, other than that LGTM.

File: receiver/azuremonitorreceiver/README.md [hunk 1]
```diff
@@ -4,10 +4,11 @@
 | Status        |           |
 | ------------- |-----------|
 | Stability     | [development]: metrics   |
-| Distributions | [contrib] |
+| Distributions | [contrib], [sumo] |
```
[the remark this answers is not in the document]
Comment (perk-sumo) [reply receiver/azuremonitorreceiver/README.md:7]: Good catch - we should have it though.

**Review by perk-sumo — APPROVED** [review]
LGTM pending one comment from @swiatekm-sumo.
Thank you @astencel-sumo! 🚀

File: receiver/azuremonitorreceiver/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (perk-sumo) [reply receiver/azuremonitorreceiver/README.md:7]: Ah, it's in `development` - so I take that back - I agree with @swiatekm-sumo we should remove it from here.

**Review by codeboten — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]

File: receiver/azuremonitorreceiver/metadata.yaml [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+  distributions: [contrib]
 
 resource_attributes:
   azuremonitor.tenant_id:
-    description:  Azure tenant ID
+    description: Azure tenant ID
     type: string
   azuremonitor.subscription_id:
-    description:  Azure subscription ID
+    description: Azure subscription ID
```
Comment (atoulme) [inline receiver/azuremonitorreceiver/metadata.yaml:14]: these are whitespace changes only, maybe a separate PR for those?

**Pull request merged**